Microsoft.Research/Contracts/WindowsBase/System.ComponentModel.ICollectionView.cs

```diff
@@ -0,0 +1,505 @@
+        //     an item within the view; otherwise, false.
+        bool MoveCurrentToPosition(int position);
+        //
+        // Summary:
+        //     Sets the item before the System.ComponentModel.ICollectionView.CurrentItem
+        //     in the view as the System.ComponentModel.ICollectionView.CurrentItem.
+        //
+        // Returns:
+        //     true if the resulting System.ComponentModel.ICollectionView.CurrentItem is
+        //     an item within the view; otherwise, false.
+        bool MoveCurrentToPrevious();
+        //
+        // Summary:
+        //     Recreates the view.
+        void Refresh();
+    }
+
+    [ContractClassFor(typeof (ICollectionView))]
+    abstract class ICollectionViewContract : ICollectionView
+    {
+        event NotifyCollectionChangedEventHandler INotifyCollectionChanged.CollectionChanged
+        {
+            add
+            {
+            }
+            remove
+            {
+            }
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            throw new NotImplementedException();
+        }
+
+        bool ICollectionView.CanFilter
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        bool ICollectionView.CanGroup
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        bool ICollectionView.CanSort
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        CultureInfo ICollectionView.Culture
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+            set
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        object ICollectionView.CurrentItem
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        int ICollectionView.CurrentPosition
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        Predicate<object> ICollectionView.Filter
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+            set
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        ObservableCollection<GroupDescription> ICollectionView.GroupDescriptions
+        {
+            get
+            {
+                Contract.Ensures(Contract.Result<ObservableCollection<GroupDescription>>() != null);
+                throw new NotImplementedException();
+            }
+        }
+
+        ReadOnlyObservableCollection<object> ICollectionView.Groups
+        {
+            get
+            {
+                Contract.Ensures(Contract.Result<ReadOnlyObservableCollection<object>>() != null);
```

Comment:
The documentation mentions that this could return null if there are no groups: https://msdn.microsoft.com/en-us/library/system.componentmodel.icollectionview.groups(v=vs.110).aspx